core_contracts/feehandler/feehandler.py

```diff
@@ -157,36 +158,82 @@ def tokenFallback(self, _from: Address, _value: int, _data: bytes) -> None:
+                                self._getTokenBalance(self.msg.sender))
 
         # Set the block for this fee processing event.
         self._last_fee_processing_block[self.msg.sender] = self.block_height
 
+    @only_owner
+    @external
+    def add_allowed_address(self, address: Address) -> None:
+        """
+        Adds address into  allowed address list
+        :param address: Address to be added
+        :return:
+        """
+        self._allowed_addresses.put(address)
+
+    @external(readonly=True)
+    def get_allowed_address(self, offset: int = 0) -> List:
+        """
+        Returns 20 allowed address
+        :param offset: Offset
+        :return:
+        """
+        start = offset
+        end = min(len(self._allowed_addresses), offset + 20) - 1
+        return [self._allowed_addresses[i] for i in range(start, end + 1)]
+
+    @external
+    def route_contract_balances(self) -> None:
+        """
+        Converts and sends fees held by the fee handler to destination contract
+        :return:
+        """
+        starting_index = self._current_allowed_addresses_index.get()
+        current_index = starting_index
+        balance = 0
+        address = None
+
+        allowed_addresses_length = len(self._allowed_addresses)
+        if not allowed_addresses_length:
+            revert(f"{TAG}: No allowed addresses.")
+
+        while not balance:
+            current_index += 1
+            if current_index == starting_index:
+                revert(f"{TAG}: Contract has no balance for any of the allowed addresses.")
+            if current_index >= allowed_addresses_length:
+                current_index = 0
+            address = self._allowed_addresses[current_index]
+            balance = self._getTokenBalance(address)
+        self._current_allowed_addresses_index.set(current_index)
+
+        try:
+            # Raises JSONDecodeError if trying to decode an empty string.
+            path = json_loads(self._routes[self.msg.sender][self._getContractAddress("baln")])
+        except:
+            path = []
+
+        # Attempt to process fees. Catch Error and report via eventlog if failure.
+        balance = self._getTokenBalance(address)
+        try:
+            if path:
+                # Use router.
+                self._transferToken(address, self._getContractAddress("router"), balance,
+                                    self._createDataFieldRouter(self._getContractAddress("dividends"), path))
+
+            else:
+                # Use dex.
+                self._transferToken(address, self._getContractAddress("dex"), balance,
+                                    self._createDataFieldDex(self._getContractAddress("baln"),
+                                                             self._getContractAddress("dividends")))
+
+        except BaseException as e:
+            revert(f'Fee conversion for {address} failed, {repr(e)}')
+            # self.FeeNotProcessed(self.msg.sender, repr(e))
```

Comment:
Looks like we will not be needing this eventlog. Let's remove it, along with the eventlog comment on line 219 and the eventlog definition at the end.

Reply:
Commented code removed along with the definition